Summary/analyze/util.py

```diff
@@ -64,7 +68,7 @@ def get_webpages(doc_home, repo_name):
         # We do not want links with "#" because they're likely redundant
         if parse.hostname is None:
             if parse.path and not parse.fragment and not parse.query:
-                url = re.match(re.compile(".+"), doc_home)[0] + parse.path
+                url = base_url + parse.path
```

Comment:
Can you explain the change a bit more to me and which examples you tested it with? Like in the example I thought the issue is with having contents.html while you do not want that?

Reply:
So for the previous code, if the base_url was  this - 'https://docs.pytest.org/en/7.2.x/contents.html#', and then the url would be this -  'https://docs.pytest.org/en/7.2.x/contents.html#how-to/assert.html' 
But we don't want the fragment, so we refactored the code to get this - 'https://docs.pytest.org/en/7.2.x/how-to/assert.html'